frontend/src/app/main/data/common.cljs

```diff
@@ -170,3 +173,65 @@
         (->> (rp/cmd! :create-team-access-request params)
              (rx/tap on-success)
              (rx/catch on-error))))))
+
+
+(defn change-team-permissions
+  [{:keys [team-id role workspace?]}]
+  (dm/assert! (uuid? team-id))
+  (dm/assert! (contains? tt/valid-roles role))
+  (ptk/reify ::change-team-permissions
+    ptk/WatchEvent
+    (watch [_ _ _]
+      (let [msg (case role
+                  :viewer
+                  (tr "dashboard.permissions-change.viewer")
+
+                  :editor
+                  (tr "dashboard.permissions-change.editor")
+
+                  :admin
+                  (tr "dashboard.permissions-change.admin")
+
+                  :owner
+                  (tr "dashboard.permissions-change.owner"))]
+        (rx/of (ntf/info msg))))
+
+    ptk/UpdateEvent
+    (update [_ state]
+      (let [route (if workspace?
+                    [:workspace-file :permissions]
+                    [:teams team-id :permissions])]
+        (update-in state route
+                   (fn [permissions]
+                     (cond
```

Comment:
I think we can extract this logic to `app.common.types.team`, the logic of changing a permission objects with a role keyword is also used on backend I think, so we can start reusing this code also.